src/common/k8s-api/kube-object.store.ts

```diff
@@ -274,10 +274,10 @@ export abstract class KubeObjectStore<T extends KubeObject> extends ItemStore<T>
 
     // update existing items
     if (merge) {
-      const namespaces = partialItems.map(item => item.getNs());
+      const uids = new Set(partialItems.map(item => item.getId()));
 
       items = [
-        ...this.items.filter(existingItem => !namespaces.includes(existingItem.getNs())),
+        ...this.items.filter(existingItem => !uids.has(existingItem.getId())),
         ...partialItems,
       ];
```

Comment:
Not obvious why this is needed?

Reply:
I'll add a comment then, but the main gist is that stores cannot assume that a given call to `mergeItems` represents all namespaces being watched. 

Specifically, a Logs tab might be viewing logs from a pod in `default` while the pods view is showing only the ones in `kube-system`